Assets/Awsim/Scripts/Entity/Environments/SmokeGenerator/SmokeParticle.cs

```diff
@@ -0,0 +1,135 @@
+using UnityEngine;
+using Awsim.Entity;
+
+namespace Awsim.Entity
+{ 
+	[RequireComponent(typeof(MeshFilter))]
+	[RequireComponent(typeof (MeshRenderer))]
+
+	/// <summary>
+	/// Smoke Particle class.
+	/// </summary>
+	public class SmokeParticle : MonoBehaviour
```

Comment:
I want to reduce the number of classes that inherit from MonoBehaviour as much as possible; can I integrate them into the SmokeGenerator class?